core/src/main/scala/caliban/parsing/VariablesUpdater.scala

```diff
@@ -16,38 +19,75 @@ object VariablesUpdater {
   ): GraphQLRequest = {
     val variableDefinitions = doc.operationDefinitions.flatMap(_.variableDefinitions)
     val updated             = req.variables.getOrElse(Map.empty).map { case (key, value) =>
-      val v = variableDefinitions.find(_.name == key).map(resolveEnumValues(value, _, rootType)).getOrElse(value)
+      val v =
+        variableDefinitions
+          .find(_.name == key)
+          .map { definition =>
+            rewriteValues(value, definition.variableType, rootType)
+          }
+          .getOrElse(value)
+
       key -> v
     }
 
     req.copy(variables = Some(updated))
   }
 
+  private def rewriteValues(value: InputValue, `type`: Type, rootType: RootType): InputValue =
+    `type` match {
+      case ListType(ofType, _) =>
+        value match {
+          case ListValue(values) =>
+            ListValue(values.map(v => rewriteValues(v, ofType, rootType)))
+          case _                 => value
+        }
+      case NamedType(name, _)  =>
+        rootType.types.get(name).map(t => resolveEnumValues(value, t, rootType)).getOrElse(value)
```

Comment:
What if this one has a list inside. It doesn't need to recurse on this function?

Reply:
Lists should be [handled here](https://github.com/ghostdogpr/caliban/blob/38daa5e892b803a4d370d4d565beeeac23dc4882/core/src/main/scala/caliban/parsing/VariablesUpdater.scala#L71-L78) -- this is more that we need to special case the top level `Type` before we get the "real" `__Type`.

Reply:
It should be covered by this test case https://github.com/ghostdogpr/caliban/blob/38daa5e892b803a4d370d4d565beeeac23dc4882/core/src/test/scala/caliban/execution/FieldArgsSpec.scala#L77-L114